src/index.js

```diff
@@ -228,26 +228,20 @@ const getIdKeys = (pid) => {
 }
 
 // Sign ipns record data
-const sign = (privateKey, value, validityType, validity, callback) => {
+const sign = (privateKey, value, validityType, validity) => {
   const dataForSignature = ipnsEntryDataForSig(value, validityType, validity)
-
-  privateKey.sign(dataForSignature, (err, signature) => {
-    if (err) {
-      return callback(err)
-    }
-    return callback(null, signature)
-  })
+  return privateKey.sign(dataForSignature)
```

Comment:
We need await here, right?

```suggestion
  return await privateKey.sign(dataForSignature)
```

Reply:
I don't think so - `privateKey.sign()` returns a Promise, which we then return from our function